service/rpc/state.go

```diff
@@ -116,6 +119,12 @@ func (h *Handler) handleSubmitPFD(w http.ResponseWriter, r *http.Request) {
 		writeError(w, http.StatusBadRequest, submitPFDEndpoint, err)
 		return
 	}
+	// sanity check nID size
+	if len(nID) != ipld.NamespaceSize {
+		writeError(w, http.StatusBadRequest, submitPFDEndpoint,
+			fmt.Errorf("namespace id must be default size %d", ipld.NamespaceSize))
+		return
+	}
```

Comment:
Same here. Maybe it will be possible to make only one check lower in the stack instead of two here.

Reply:
https://github.com/celestiaorg/celestia-app/pull/418